dataactvalidator/filestreaming/loaderUtils.py

```diff
@@ -25,85 +28,62 @@ def compareRecords (recordA,recordB, fields) :
-            # Map fieldnames to attribute names
-            attributeNames = []
-            for field in fieldNames:
-                if field in fieldMap:
-                    attributeNames.append(fieldMap[field])
-                    if fieldMap[field] in fieldOptions and "skip_duplicates" in fieldOptions[fieldMap[field]]:
-                        # Create empty dict for this field
-                        valuePresent[fieldMap[field]] = {}
-                else:
-                    raise KeyError("".join(["Found unexpected field ", str(field)]))
-            # Check that all fields are present
-            for field in fieldMap:
-                if not field in fieldNames:
-                    raise ValueError("".join([str(field)," is required for loading table ", str(type(model))]))
-            # Open DictReader with attribute names
-            reader = csv.DictReader(csvfile,fieldnames = attributeNames)
-            # For each row, create instance of model and add it
-            for row in reader:
-                skipInsert = False
-                for field in fieldOptions:
-                    if row[field] is None:
-                        # If field is empty set to an empty string
-                        row[field] = ""
-                    # For each field with options present, modify according to those options
-                    options = fieldOptions[field]
-                    if "strip_commas" in options:
-                        # Remove commas from numeric fields
-                        row[field] = row[field].replace(",","")
-                        if row[field] == "":
-                            # If empty, set to 0
-                            row[field] = "0"
-                    if "pad_to_length" in options:
-                        padLength = options["pad_to_length"]
-                        row[field] = Validator.padToLength(row[field],padLength)
-                    if "skip_duplicates" in options:
-                        if row[field] is None or len(row[field].strip()) == 0 or row[field] in valuePresent[field]:
-                            # Value not provided or already exists, skip it
-                            skipInsert = True
-                        else:
-                            # Insert new value
-                            valuePresent[field][row[field]] = True
-                record = model(**row)
-                if not skipInsert:
-                    try:
-                        interface.session.merge(record)
-                        interface.session.commit()
+        # toss out blank rows
+        data.dropna(inplace=True, how='all')
+
+        # Fix up cells that have spaces instead of being empty.
+        # Set the truly empty cells to None so they get inserted to db as NULL
+        # TODO: very ugly function below...is there a better way?
+        data.applymap(lambda x: str(x).strip() if len(str(x).strip()) else None)
```

Comment:
Thanks for taking a look at this. The applymap is pretty expensive to do on these large dataframes, so I'm hesitant to do it twice. I've been through the Pandas stuff and couldn't find anything magic that will handle the case where a cell has spaces. I did play with setting whitespace cells to NaN when the file was read, but that cascading into some other issues.

In conclusion--it would be good to make this process less fragile but am hesitant to pull this thread.